Update the tested Standard C++ APIs for `libc++` (#2080)
uxlfoundation/oneDPL: documentation/library_guide/api_for_sycl_kernels/tested_standard_cpp_api.rst

``std::truncf``                       Tested     Tested     Tested
===================================== ========== ========== ==========

> These tests were done for the following SYCL compilers and versions of the standard C++ library:

Reviewer: As far as I understand, a compiler served as a tool for testing a standard library. E.g. an additional, AdaptiveCPP, was used to test libc++, because Intel(R) oneAPI DPC++/C++ Compiler is linked against libstdc++. 

The suggestion shifts focus to the libraries:

```suggestion
The testing was conducted for the following versions of the C++ standard libraries, with the help of the listed compilers.:
```

It may also make sense to put the compiler into the last column and name it as "SYCL Compiler Used".

Author: Done.

============================================= ============================================= ==================================

> Standard C++ Library                          Standard C++ Library Version                  SYCL Compiler Used

Reviewer: ```suggestion
C++ Standard Library                          C++ Standard Library Version                  SYCL Compiler Used
```

> SYCL Compiler                      Standard C++ Library                          Standard C++ Library Version

Reviewer: I suggest putting the compiler into the last column and name it as "SYCL Compiler Used".

https://github.com/uxlfoundation/oneDPL/pull/2080#discussion_r2075161559 describes why.

Author: Done

These tests were done for the following versions of the standard C++ library:

============================================= ============================================= ==========================
libstdc++ (GNU)                               Provided with GCC* 8.4.0, GCC 9.3.0,
                                              GCC 11.4.0, GCC 13.2.0
--------------------------------------------- --------------------------------------------- --------------------------

> libc++ (LLVM)                                 Provided with Clang* 20.1                     AdaptiveCpp compiler

Reviewer: Let's write only "AdaptiveCPP" without "compiler".
It is mentioned as "AdaptiveCPP" everywhere in it's documentation.

I also recommend naming the newly added column.

> libc++ (LLVM)                                 Provided with Clang* 20.1                     AdaptiveCpp compiler

Author: Done

libstdc++ (GNU)                               Provided with GCC* 8.4.0, GCC 9.3.0,
                                              GCC 11.4.0, GCC 13.2.0
--------------------------------------------- ---------------------------------------------
libc++ (LLVM)                                 Provided with Clang* 20.1
--------------------------------------------- ---------------------------------------------
Microsoft Visual C++* (MSVC) Standard Library Provided with Microsoft Visual Studio 2019

>                                               and Microsoft Visual Studio 2022.

Reviewer 2: I think we need to extend the table to also indicate which SYCL compiler(s) we used for testing.

Author: Extended.
